packages/program/src/functions/base-invocation-scope.ts

```diff
@@ -352,9 +355,10 @@ export class BaseInvocationScope<TReturn = any> {
     const provider = this.getProvider();
 
     const transactionRequest = await this.getTransactionRequest();
-
     const { maxFee } = await this.getTransactionCost();
 
+    await this.setDefaultTxParams();
+
     await this.fundWithRequiredCoins(maxFee);
```

Comment:
```suggestion
    await this.setDefaultTxParams();

    const { maxFee } = await this.getTransactionCost();
    await this.fundWithRequiredCoins(maxFee);

    const transactionRequest = await this.getTransactionRequest();
```

Same as in `call` suggestion.